shared/hooks/useSearch.ts

```diff
@@ -22,6 +23,18 @@ export const prepareSearchDocuments = (
 // Modifying single documents or replacing documents with a same length array will therefore NOT trigger an update
 const useSearch = (documents: SearchResult[], query: string): SearchResult[] => {
   const [indexing, setIndexing] = useState(false)
+  const [debouncedQuery, setDebouncedQuery] = useState(query)
+
+  useEffect(() => {
+    const debounceQueryTimeout = setTimeout(() => {
+      setDebouncedQuery(debouncedQuery)
+    }, DEBOUNCED_QUERY_TIMEOUT)
+
+    return () => {
+      clearTimeout(debounceQueryTimeout)
+    }
+  }, [debouncedQuery])
```

Comment:
:upside_down_face: Could you perhaps move this below the mini search initialization/useState? Feels more like the correct order (initialization first, effects and behavior second).